### DISCUSSION TITLE
Setting read method for opcua client.

### DISCUSSION THREAD

**Discussion by GauravMundhra** [FreeOpcUa/python-opcua#1312] [Q&A]
Hello,

There are different read methods for OPC UA clients i.e subscription, synchronous, asynchronous.

How can I set subscription as a read method when I connect through python-opcua?

Thank you!

**Comment by AndreasHeine** [comment]
read and write are part of the AttributeServiceSet of opcua (https://reference.opcfoundation.org/Core/docs/Part4/5.10.1/)
a subscription is part of the SubscriptionServiceSet of opc ua (https://reference.opcfoundation.org/Core/docs/Part4/5.13.1/ + MonitoredItemServiceSet https://reference.opcfoundation.org/Core/docs/Part4/5.12.1/)

so if you dont want to read (poll) data you can create a subscription and subscribe to variables and properties to get a notification if they change!

https://github.com/FreeOpcUa/python-opcua/blob/master/examples/client-example.py
https://github.com/FreeOpcUa/python-opcua/blob/3012acf9047aad3acc2674e9008f0d1eee680c25/examples/client-example.py#L78

**Reply by GauravMundhra** [reply]
Thank you for your reply.

Sir, when I try to retrieve data from python opcua it is getting hold to previous value and does not update.
when I connect it through Cognet Data Hub and set read method as subscription. Values update both in CognetDataHub and python output and if I close Cognet Data Hub, again in python output values hold and does not update.

Can you help with this.

**Reply by AndreasHeine** [reply]
unfortunatly i dont know cognet so i am not able to help on this side!
